RFC-011-SWAP-Omni-Layer-basic.md

```diff
@@ -0,0 +1,329 @@
+Hence, (2) can be used as a fallback.
+
+The simple send MUST NOT be done to the direct P2SH address of the HTLC.
+
+This choice has been made to:
+- limit the number of addresses to watch, facilitating implementation
+- avoid defining the basic HTLC twice (once, in P2SH format, once in P2WSH format)
+- follow the common strategy for segwit backward compatibility (nesting in P2SH)
+
+### Inputs & Outputs format to spend the HTLC
+
+This section describes specific inputs & outputs to construct an Omni Layer transaction for the *redeeming* and *refunding* of an HTLC.
+This is to facilitate the constructions and filtering of such transactions for the purpose of a basic swap.
+
+To construct a transaction to spend from the HTLC, the following information is needed:
+
+| Variable           | Description                                                                                                 |
+|:---                |:---                                                                                                         |
+| `htlc_utxo`        | UTXO locked by the HTLC, created with the deployment transaction                                            |
+| `to_address`       | Address to which the Omni Layer token is sent out of the HTLC                                               |
+| `change_address`   | Address to which the Bitcoin change will be sent upon redeeming; MUST be different than `to_address`        |
+| `change_utxo`      | UTXO locked by `change_address` that owns some Bitcoin asset to pay for the mining fee                      |
+| `init_change_sat`  | Amount of Satoshis owned by `change_utxo` before the redeem                                                 |
+| `mining_sat`       | The amount of Satoshi to be spent on mining fees                                                            |
+| `omni_data`        | The Omni Layer data that transfers the Omni Layer Asset as described in [Omni Layer data](#omni-layer-data) |
+
+ The Bitcoin transaction MUST be constructed with the following inputs, outputs and values.
+ The order of the inputs and outputs MUST be as described.
+ The *assigned Bitcoin value* on the inputs SHOULD be as described.
+
+ | Inputs (assigned Bitcoin value)      | Outputs (assigned Bitcoin value)                     |
+ |:---                                  |:---                                                  |
+ | 1. `htlc_utxo` (`min_sat`)           | 1. `to_address` (`min_sat`)                          |
+ | 2. `change_utxo` (`init_change_sat`) | 2. `change_address` (`init_change_sat - mining_sat`) |
+ |                                      | 3. `OP_RETURN omni_data` (`0`)                       |
+
+The implementor MAY include output (2) `change_address` or MAY decide to omit it and transfer the Bitcoin change to `to_address`.
+if output (2) `change_address` is present then input (2) `change_utxo` MUST be included to ensure the Omni Layer Assets are sent to output (1) `to_address`.
```

Comment:
I think the redeemer and refunder will always need to provide a `change_utxo` input as the `htlc_utxo` won't have enough btc to cover for the transaction fee. Can you confirm this?

Reply:
yes good catch

Reply:
I added a sentence